lib/systemu.rb

```diff
@@ -244,7 +250,7 @@ def tmpdir d = Dir.tmpdir, max = 42, &b
       begin
         Dir.mkdir tmp
       rescue Errno::EEXIST
-        raise if i >= max
+        raise Error if i >= max
```

Comment:
I wouldn't mask the error here. The exception on a failed mkdir will let you know what is wrong, here you're hiding the error (well, shoving it in the backtrace)

Reply:
Ahh I didn't realize that calling `raise` with no arguments would raise the exception in `$!` if it is not nil rather than just raising a `RuntimeError`.  (Thought calling it with no arguments always raised a `RuntimeError`, which is why I replaced it with the systemu specific runtime error.)